server/stycoon/commands.go

```diff
@@ -14,7 +15,7 @@ func ParseCommands(s string) (map[string]Command, error) {
 	return c, err
 }
 
-func (game *Game) ProcessCommands(commands map[string]Command) map[string]string {
+func (game *Game) ProcessCommands(commands map[string]Command, user LoggedUser) map[string]string {
 	res := map[string]string{}
 	for strId, command := range commands {
```

Comment:
performance issues here: it is likely that players will be sending commands for all of their ships every single tick. considering that each player will have possibly hundreds of ships, having a separate sql query for each command becomes significant bottleneck. even more with two sql queries per command (test object owner and replace the command).
correct approach is to fetch all owners in single query and cache it (for the duration of this single http request).
similarly collect all commands and replace them into the database all in a single sql query.

all that said, it is preferable to have this merged as is asap and fix performance later.

Reply:
Very good point. Thank you, changed for fetching all owned ships and then just querying that set for each of the commands.